src/app/backend/resource/deployment/rollout.go

```diff
@@ -39,17 +45,45 @@ type RolloutSpec struct {
 	Revision string `json:"revision"`
 }
 
+// HistoryList contains a list of Replica Sets in the deployment.
+type HistoryList struct {
+	ListMeta api.ListMeta `json:"listMeta"`
+
+	// Basic information about resources status on the list.
+	Status common.ResourceStatus `json:"status"`
+
+	// Unordered list of Replica Sets.
+	HistoryList []History `json:"history"`
+
+	// List of non-critical errors, that occurred during resource retrieval.
+	Errors []error `json:"errors"`
+}
+
+// History is a presentation layer view of Kubernetes Replica Set resource for history.
+type History struct {
```

Comment:
This kind of breaks our pattern of resource being a Kubernetes resource. It is then used in the frontend as a resource also. I'd have to rethink how to incorporate that into our patterns.

Reply:
Yes, these are defined in frontend also.
To create dialog contents and submission endpoint URL, the action on history row needed to pass deployment name and namespace to dialog. So I defined new type History that have ReplicaSet and parent resource information. Could you let me know if you have better idea about this?

Reply:
I'll think about how we could add this without breaking our patterns. Resource should be mappable to a real Kubernetes resource. History/HistoryList is a very abstract thing.